### PR TITLE
chore: update benchmark-compare and github action

### PR DESCRIPTION
**Pull request by pi0**
I was trying to test changes in #163 in my fork, it seemes `benchmark-compare` is stalled and not working and containes lots of duplicate logic. So I tried to rewrite and improve it. Also updated github action accordingly to use node LTS, new `compare -u` script to update readme and [github-actions-x/commit](https://github.com/github-actions-x/commit) to simplify commit and push step. Tried action and script in [my fork](https://github.com/pi0/benchmarks) with a subset of `benchmarks/`.

Note: Percentage flavor is removed since wasn't sure this PR and that feature is still wanted or not

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: benchmark-compare.js
Comment (RafaelGSS) [inline benchmark-compare.js:18]: How this action should work? I've tested locally but seems not to work.

```sh
rafaelgss@xxx:~/repos/os/benchmarks$ node ./benchmark compare -m
? What's your first pick? express
? What's your second one? fastify

 Both are awesome but fastify is 364% faster than express
 • fastify request average is 29113.6
 • express request average is 6274.43
```

Comment (pi0) [reply benchmark-compare.js:18]: `benchmark compare -t [-m]` renders comparison table while the default behavior is interactive mode (as it was)

In github action, `compare -u` is used which updates `README.md` (logic moved from GitHub action steps)

**Review by RafaelGSS — APPROVED** [review]
I've tested locally and seems to work. But, I'll wait for another approval from the team since I didn't work in this repo for a while.

**Comment by mcollina** [comment]
go ahead and land @RafaelGSS, I won't have time to review. If it works locally for you it's good.

**Comment by RafaelGSS** [comment]
Thanks @pi0!

**Pull request merged**
